snap/revision.go

```diff
@@ -56,6 +56,18 @@ func (r Revision) MarshalJSON() ([]byte, error) {
 	return []byte(`"` + r.String() + `"`), nil
 }
 
+func (r *Revision) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var s string
+	if err := unmarshal(&s); err != nil {
+		return err
+	}
+	return r.UnmarshalJSON([]byte(`"` + s + `"`))
```

Comment:
I like the idea, I would like to do it in a followup branch though, I really want to get this branch out to unblock image building.